x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -247,6 +247,41 @@ public void testComparisonAgainstColumns() {
         assertEquals("Line 1:43: Comparisons against fields are not (currently) supported; offender [int] in [>]", ex.getMessage());
     }
 
+    public void testComparisonAgainstColumnsSameSide() {
+        LogicalPlan p = plan("SELECT some.string FROM test WHERE int - float > 0");
```

Comment:
Just an observation: `...WHERE IIF(int - float > 0, true, false)` is currently scripted fine. Might be the case that we could suggest an alternative or check the validity.

Reply:
The use of multiple fields inside the same function is still permitted. The IIF query you showed will still be allowed.

Reply:
I think we could look at some point at this inconsistency too. Maybe later with better Painless support.

Reply:
Yes, there's a wider applicability here. In the same ballpark as https://github.com/elastic/elasticsearch/issues/34400.